VideoLocker/src/main/java/org/edx/mobile/view/MyRecentVideosFragment.java

```diff
@@ -206,33 +215,48 @@ private void setCheckBoxVisible(boolean visible) {
         }
     }
 
-    private void addToRecentAdapter(View view) {
+    private void addToRecentAdapter(final View view) {
         if (adapter == null) {
             return;
         }
         logger.debug("reloading adapter...");
-        String selectedId = adapter.getVideoId();
-        ArrayList<SectionItemInterface> list = environment.getStorage().getRecentDownloadedVideosList();
-        if (list != null) {
-            adapter.clear();
-            for (SectionItemInterface m : list) {
-                adapter.add(m);
-            }
-            logger.debug("reload done");
-        }
-        if (adapter.getCount() <= 0) {
-            hideDeletePanel(view);
-        }
-        videoListView.setOnItemClickListener(adapter);
-        if (selectedId != null) {
-            adapter.setVideoId(selectedId);
+        final String selectedId = adapter.getVideoId();
+
+
+        if (getRecentDownloadedVideosTask != null) {
+            getRecentDownloadedVideosTask.cancel(true);
         }
-        notifyAdapter();
-        // Refresh the previous and next buttons visibility on the video
-        // player if a video is playing, based on the new data set.
-        if (playerFragment != null) {
-            playerFragment.setNextPreviousListeners(getNextListener(), getPreviousListener());
+        else {
+            getRecentDownloadedVideosTask = new GetRecentDownloadedVideosTask(getActivity()) {
+                @Override
+                protected void onSuccess(List<SectionItemInterface> list) throws Exception {
+                    super.onSuccess(list);
+                    if (list != null) {
```

Comment:
There's no reason `IStorage.getRecentDownloadedVideosList` to ever return `null`. Can we make it return an empty list and remove this null-check?

(And preferably add a `@NonNull` annotation to the interface to enforce that)